geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegion.java

```diff
@@ -1221,24 +1222,36 @@ public void addGatewaySenderId(String gatewaySenderId) {
   }
 
   public void updatePRConfigWithNewSetOfGatewaySenders(Set<String> gatewaySendersToAdd) {
-    PartitionRegionHelper.assignBucketsToPartitions(this);
-    PartitionRegionConfig prConfig = this.prRoot.get(getRegionIdentifier());
-    prConfig.setGatewaySenderIds(gatewaySendersToAdd);
-    updatePRConfig(prConfig, false);
+    final RegionLock rl = getRegionLock();
+    try {
+      rl.lock();
```

Comment:
I generally prefer to the lock() call *outside* the try finally - otherwise we will call unlock even if lock fails.

Reply:
Done